clang/test/SemaSYCL/sycl-varargs-cconv.cpp

```diff
@@ -1,5 +1,12 @@
 // RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only -x c++ %s
 
+extern int __spirv_ocl_printf(const char *__format, ...);
+
+template<typename... Args>
+int printf(const char *__format, Args... args) {
+  return __spirv_ocl_printf(__format, args...);
```

Comment:
We are checking only `__spirv_ocl_printf` function. Don't you know why printf wrapper is not diagnosed?

Reply:
This `printf` wrapper doesn't have variadic arguments (but it has variadic template arguments, which is different thing)

Reply:
Shouldn't we also diagnose variadic template arguments?

Reply:
I think the diagnosis/restriction for variadics macros/function is due to OpenCL C requiring it, but OpenCL C  has no templates so the restriction never really covered those.

I think that function variadics are also different in that they require some C/C++ runtime support, whereas template parameter packs/variadics will inevitably be instantiated if used and the end result will look not so different from a normal function that someone handwrote without variadics.

That's my very likely flawed understanding of it anyway, if I'm incorrect I'd be interested in the correct reasoning (in particular the historical reason for the restriction in OpenCL C)!

Reply:
> whereas template parameter packs/variadics will inevitably be instantiated if used and the end result will look not so different from a normal function (or set of) that someone handwrote without variadics.

This is my understanding as well, variadic template is instantiated into a bunch of function overloads with different set of arguments, one calling another one - so, nothing wrong from the SYCL spec prospective

Reply:
okay